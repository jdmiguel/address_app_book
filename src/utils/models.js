import PropTypes from 'prop-types';

export const iconModel = PropTypes.shape({
  src: PropTypes.string,
  alt: PropTypes.string,
});
