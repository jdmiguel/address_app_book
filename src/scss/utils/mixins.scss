@mixin flexContainer($mainAxisAlign:center,$crossAxisAlign:center,$direction:row,$wrap:nowrap){
  display: flex;
  justify-content: $mainAxisAlign;
  align-items: $crossAxisAlign;
  flex-direction: $direction;
  flex-wrap: $wrap;
}

@mixin font($size,$weight,$color:$black) {
  font-size: $size;
  font-weight: $weight;
  color: $color;
}

@mixin transition($property,$duration,$easing:ease-out,$delay:0) {
  -webkit-transition-property: $property;
  -webkit-transition-duration: $duration;
  -webkit-transition-timing-function: $easing;
  -webkit-transition-delay: $delay;

  -khtml-transition-property: $property;
  -khtml-transition-duration: $duration;
  -khtml-transition-timing-function: $easing;
  -khtml-transition-delay: $delay;

  -moz-transition-property: $property;
  -moz-transition-duration: $duration;
  -moz-transition-timing-function: $easing;
  -moz-transition-delay: $delay;

  -ms-transition-property: $property;
  -ms-transition-duration: $duration;
  -ms-transition-timing-function: $easing;
  -ms-transition-delay: $delay;

  transition-property: $property;
  transition-duration: $duration;
  transition-timing-function: $easing;
  transition-delay: $delay;
}