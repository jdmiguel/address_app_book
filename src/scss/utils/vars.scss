// paths
$imgPath: "../assets/img";

// colors
$white: #FFFFFF;
$darkWhite: #e5e4e9;
$lightGrey: #c0bcbc;
$grey: #5e5e5e;
$black: #221e1f;
$primary: #A66885;
$primaryDark: #734359;
$primaryLight: #F2C1AE;
