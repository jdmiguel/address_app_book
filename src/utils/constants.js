import settingsIconSrc from '../assets/img/settings.svg';
import homeIconSrc from '../assets/img/home.svg';
import placeIconSrc from '../assets/img/place.svg';
import exploreIconSrc from '../assets/img/explore.svg';
import phoneIconSrc from '../assets/img/phone.svg';
import warningIconSrc from '../assets/img/warning.svg';
import searchingIconSrc from '../assets/img/searching.svg';
import checkIconSrc from '../assets/img/check.svg';

import spainFlagSrc from '../assets/img/spain-flag.png';
import franceFlagSrc from '../assets/img/france-flag.png';
import switzerlandFlagSrc from '../assets/img/switzerland-flag.png';
import unitedKingdomFlagSrc from '../assets/img/united-kingdom-flag.png';

// icons config
export const settingsIcon = {
  src: String(settingsIconSrc),
  alt: 'settings',
};
export const homeIcon = { src: String(homeIconSrc), alt: 'home' };
export const modalIcons = {
  firstIcon: { src: String(placeIconSrc), alt: 'address' },
  secondIcon: { src: String(exploreIconSrc), alt: 'location' },
  thirdIcon: { src: String(phoneIconSrc), alt: 'phone' },
};
export const warningIcon = { src: String(warningIconSrc), alt: 'warning' };
export const searchingIcon = {
  src: String(searchingIconSrc),
  alt: 'address',
};
export const checkIcon = { src: String(checkIconSrc), alt: 'check' };

// literals
export const literals = {
  logo: 'ADDRESS BOOK',
  footer: {
    text: 'This is a coding exercise for Sherpany company',
    link: {
      settingsText: 'Go to Settings',
      settingsLink: '/settings',
      homeText: 'Back to Home',
      homeLink: '/',
    },
    warning:
      'While finder is active, more users will not be loaded by scrolling',
  },
  loadingText: 'Loading...',
  settingsPageTitle: 'Set one of these nationalities as default',
  finderPlaceholder: 'Start typing to filter by user...',
  userAltImg: 'user image',
  errorText:
    'Sorry but the last request has failed, try again by reloading the page ;)',
  endUsersText: 'END OF USERS CATALOG',
};

// nationalities config
export const nationalities = {
  swiss: {
    id: 'ch',
    src: String(switzerlandFlagSrc),
    text: 'Swiss',
  },
  spanish: {
    id: 'es',
    src: String(spainFlagSrc),
    text: 'Spanish',
  },
  french: {
    id: 'fr',
    src: String(franceFlagSrc),
    text: 'French',
  },
  british: {
    id: 'gb',
    src: String(unitedKingdomFlagSrc),
    text: 'British',
  },
};

// scroll utils
export const scrollTrigger = 50;
export const scrollFactor = 1.6;
export const maxRequests = 11;
export const batchUsersByRequest = 100;

// defaults
export const defaultImgSrc = 'https://randomuser.me/api/portraits/women/52.jpg';
export const defaultCardData = {
  cardFirstLine: 'Alba Caballero',
  cardSecondLine: 'purpleelephant787',
  cardThirdLine: 'alba.caballero@example.com',
};
export const defaultNationalityText = 'Spanish';
export const defaultModalData = {
  imgSrc: defaultImgSrc,
  firstText: 'Alba Caballero',
  secondText: 'purpleelephant787',
  thirdText: 'alba.caballero@example.com',
  fourthText: 'Avenida de Castilla',
  fifthText: '4639',
  sixthText: 'Valencia',
  seventhText: 'Comunidad de Madrid',
  eighthText: '56271',
  ninethText: '942-033-129',
  tenthText: '689-284-149',
};
