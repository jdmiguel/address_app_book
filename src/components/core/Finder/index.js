import React from 'react';
import PropTypes from 'prop-types';

import { literals, searchingIcon, checkIcon } from '../../../utils/constants';

const Finder = ({ onChange, isSearching, isMatched }) => (
  <div className="finder">
    <input
      placeholder={literals.finderPlaceholder}
      onChange={(event) => onChange(event.target.value)}
      autoComplete="nope"
    />
    {isSearching && <img src={searchingIcon} alt="icon" />}
    {isMatched && <img src={checkIcon} alt="icon" />}
  </div>
);

Finder.propTypes = {
  onChange: PropTypes.func.isRequired,
  isSearching: PropTypes.bool,
  isMatched: PropTypes.bool,
};

Finder.defaultProps = {
  isSearching: false,
  isMatched: false,
};

export default Finder;
