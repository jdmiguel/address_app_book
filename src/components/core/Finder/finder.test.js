import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import Finder from '.';

import { literals } from '../../../utils/constants';

describe('Component: Finder', () => {
  it('should render and call onChange function', () => {
    const { container } = render(<Finder />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with search icon and call onChange function', () => {
    const handleChange = jest.fn();
    const { container, getByPlaceholderText } = render(
      <Finder isSearching onChange={handleChange} />,
    );

    expect(container.firstChild).toMatchSnapshot();

    fireEvent.change(getByPlaceholderText(literals.finderPlaceholder), {
      target: { value: 'a' },
    });
    // mock function should be called
    expect(handleChange).toHaveBeenCalled();
  });

  it('should render with matched icon', () => {
    const { container } = render(<Finder isMatched />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
