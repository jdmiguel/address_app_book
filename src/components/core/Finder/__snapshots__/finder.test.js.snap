// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Finder should render and call onChange function 1`] = `
<div
  class="finder"
>
  <input
    autocomplete="nope"
    placeholder="Start typing to filter by user..."
  />
</div>
`;

exports[`Component: Finder should render with matched icon 1`] = `
<div
  class="finder"
>
  <input
    autocomplete="nope"
    placeholder="Start typing to filter by user..."
  />
  <img
    alt="check"
    src="function svgrLoader(source) {
  const callback = this.async();

  const _ref = (0, _loaderUtils.getOptions)(this) || {},
        {
    babel = true
  } = _ref,
        options = _objectWithoutPropertiesLoose(_ref, [\\"babel\\"]);

  const readSvg = () => new Promise((resolve, reject) => {
    this.fs.readFile(this.resourcePath, (err, result) => {
      if (err) reject(err);
      resolve(result);
    });
  });

  const previousExport = (() => {
    if (source.toString('utf-8').startsWith('export ')) {
      return source;
    }

    const exportMatches = source.toString('utf-8').match(/^module.exports\\\\s*=\\\\s*(.*)/);
    return exportMatches ? \`export default \${exportMatches[1]}\` : null;
  })();

  const tranformSvg = svg => (0, _core2.default)(svg, options, {
    caller: {
      name: '@svgr/webpack',
      previousExport,
      defaultPlugins: [_pluginSvgo.default, _pluginJsx.default]
    },
    filePath: this.resourcePath
  }).then(jsCode => {
    if (!babel) return jsCode;
    return (0, _core.transformAsync)(jsCode, babelOptions).then(({
      code
    }) => code);
  }).then(result => callback(null, result)).catch(err => callback(err));

  if (previousExport) {
    readSvg().then(tranformSvg);
  } else {
    tranformSvg(source);
  }
}"
  />
</div>
`;

exports[`Component: Finder should render with search icon and call onChange function 1`] = `
<div
  class="finder"
>
  <input
    autocomplete="nope"
    placeholder="Start typing to filter by user..."
  />
  <img
    alt="address"
    src="function svgrLoader(source) {
  const callback = this.async();

  const _ref = (0, _loaderUtils.getOptions)(this) || {},
        {
    babel = true
  } = _ref,
        options = _objectWithoutPropertiesLoose(_ref, [\\"babel\\"]);

  const readSvg = () => new Promise((resolve, reject) => {
    this.fs.readFile(this.resourcePath, (err, result) => {
      if (err) reject(err);
      resolve(result);
    });
  });

  const previousExport = (() => {
    if (source.toString('utf-8').startsWith('export ')) {
      return source;
    }

    const exportMatches = source.toString('utf-8').match(/^module.exports\\\\s*=\\\\s*(.*)/);
    return exportMatches ? \`export default \${exportMatches[1]}\` : null;
  })();

  const tranformSvg = svg => (0, _core2.default)(svg, options, {
    caller: {
      name: '@svgr/webpack',
      previousExport,
      defaultPlugins: [_pluginSvgo.default, _pluginJsx.default]
    },
    filePath: this.resourcePath
  }).then(jsCode => {
    if (!babel) return jsCode;
    return (0, _core.transformAsync)(jsCode, babelOptions).then(({
      code
    }) => code);
  }).then(result => callback(null, result)).catch(err => callback(err));

  if (previousExport) {
    readSvg().then(tranformSvg);
  } else {
    tranformSvg(source);
  }
}"
  />
</div>
`;
